cirq-core/cirq/optimizers/merge_interactions.py

```diff
@@ -181,3 +203,54 @@ def _flip_kron_order(mat4x4: np.ndarray) -> np.ndarray:
         for j in range(4):
             result[order[i], order[j]] = mat4x4[i, j]
     return result
+
+
+class MergeInteractions(MergeInteractionsAbc):
+    """Combines series of adjacent one and two-qubit gates operating on a pair
```

Comment:
I think it's also worth adding a comment here and below that it works only for non-parameterized gates.

Reply:
Done.